src/udev/udev-event.c

```diff
@@ -407,6 +407,7 @@ size_t udev_event_apply_format(struct udev_event *event,
         }
 
 out:
+        assert(l >= 1);
```

Comment:
Was this line left here on purpose? If so, would it be better to put it at the top of `udev_event_apply_format` and also add `if (l == 0) return 0` in case all the assertions are excluded during the compilation?

Reply:
Yes, on purpose. I want it there to verify that the length calculations were correct, and putting it in top wouldn't work for this purpose. Likewise returning early wouldn't work: we don't want to skip the termination of the string.

Reply:
I'm not sure that the string should be terminated when `0` passed to `udev_event_apply_format` as the size of the destination string. On the other hand, `0` has never been passed and most likely won't be passed in the future, so in practice it probably makes sense to ignore that.